### PR TITLE
Feat/x search

### PR DESCRIPTION
**Pull request by seunghwako**
# Test before submitting

Before submitting a Pull Request, please test your code. If you created a new created a new feature, then create the test function, too. Please refer to [this document](https://github.com/wrtnio/connectors/blob/main/CONTRIBUTING.md).

Put the name of the test function of the service you created in the command below.
This is a regular expression that returns all tests containing the character string, so if there are several test codes, you can put a keyword including all of the test codes and turn them.
If you omit the include statement and operate the test, all the test codes we write will work, and myriad errors will occur in your code without environmental variables.

```ts
npm run build && npm run test -- --include TEST_FUNCTION_NAME
```

Of course you won't be able to provide us with your environmental variables, so we'll replace them with our environmental variables to run the program and merge it depending on whether it passes or not.

# Purpose

Please explain to our developers what the features you modified or developed are for.

# Check List

- [ ] Passing an existing or newly created test
- [ ] sufficient summary and description for LLM to understand
- [ ] Does it ensure backward compatibility with existing features, or does the modifications not affect pre-modification request, response type?

### CODE REVIEW

File: src/api/structures/connector/x/IX.ts [hunk 1]
```diff
@@ -303,4 +303,137 @@ export namespace IX {
...
+    maxResults: number & tags.Type<"int32"> & tags.Default<10>;
+
+    /**
+     * Determines the sort order.
+     * - recency: newest
+     * - relevancy: relevance
+     *
+     * @title Sort order
+     */
+    sort_order: "recency" | "relevancy";
```
Comment (kakasoo) [inline src/api/structures/connector/x/IX.ts:358]: `typia.tags.Constant`

File: src/api/structures/connector/x/IX.ts [hunk 2]
```diff
@@ -303,4 +303,137 @@ export namespace IX {
...
+     */
+    tweet_link: string & tags.Format<"iri">;
+  }
+
+  export interface IGenerelTweetOutput {
+    id: string;
+    author_id: string;
+    text: string;
+    edit_history_tweet_ids: string[];
+  }
```
Comment (kakasoo) [inline src/api/structures/connector/x/IX.ts:438]: 주석 누락.

File: src/providers/connector/x/XProvider.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ import { RagProvider } from "../rag/RagProvider";
 import typia, { tags } from "typia";
 import { v4 } from "uuid";
 import { retry } from "../../../utils/retry";
+import { use } from "dd-trace";
```
Comment (kakasoo) [inline src/providers/connector/x/XProvider.ts:19]: 이건 어디에 쓰고 있나요? 잘못 import된 거라면 지우죠.

**Review by kakasoo — COMMENTED** [review]
코멘트 남긴 것만 해결하고 merge 해주세요~

File: src/api/structures/connector/x/IX.ts (hunk 2 above)
Comment (seunghwako) [reply src/api/structures/connector/x/IX.ts:438]: 이건 LLM 동작에 크게 의미 없이 api호출 결과를 만든거라서 굳이 interface로 안 빼고 provider로 옮길게요

**Pull request merged**